src/router.ts

```diff
@@ -12,7 +12,18 @@ const routes = {
     projectDashboard: ({ id }: { id: string }) => `/project-dashboard/${id}`,
     awardNumberDashboard: ({ id }: { id: string }) => `/award-number-dashboard/${id}`,
     countryDashboard: ({ id }: { id: string }) => `/country-dashboard/${id}`,
-    dataApproval: ({ id }: { id: string }) => `/data-approval/${id}`,
+    dataApproval: ({
+        id,
+        dataSetType,
+        period,
+    }: {
+        id: string;
+        dataSetType?: string;
+        period?: string;
+    }) =>
+        dataSetType && period
+            ? `/data-approval/${id}/${dataSetType}/${period}`
+            : `/data-approval/${id}`,
```

Comment:
A bit too much code for inline function, I've moved it to a separate `function`.